File: example/src/androidTest/java/com/example/cacheexample/UUIDHexGeneratorTest.java
package com.example.cacheexample;

import com.androidyuan.libcache.core.UUIDHexGenerator;

import org.junit.Assert;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

public class UUIDHexGeneratorTest {

    private static final String TAG = "UUIDHexGeneratorTest";

    private final UUIDHexGenerator generator = new UUIDHexGenerator();

    @Test
    public void test() {
        List<String> uuidList = new ArrayList<>();

        for (int index = 0; index < 10000; index++) {
            uuidList.add(generator.generate());
        }

        for (int i = 0; i < uuidList.size(); i++) {
            for (int j = i + 1; j < uuidList.size(); j++) {
                Assert.assertNotEquals(uuidList.get(i), uuidList.get(j));
            }
        }
    }


}

File: example/src/androidTest/java/com/example/cacheexample/UUIDHexGeneratorTest.kt
package com.example.cacheexample

import com.androidyuan.libcache.core.UUIDHexGenerator
import org.junit.Assert
import org.junit.Test
import java.util.*

/**
 * This class is used to test [UUIDHexGenerator] generate ID is unique in the world.
 */
class UUIDHexGeneratorTest {
    val generator = UUIDHexGenerator()

    @Test
    fun test() {
        val uuidList: MutableList<String> = ArrayList()
        for (index in 0..9999) {
            uuidList.add(generator.generate())
        }
        for (i in uuidList.indices) {
            for (j in i + 1 until uuidList.size) {
                Assert.assertNotEquals(uuidList[i], uuidList[j])
            }
        }
    }
}